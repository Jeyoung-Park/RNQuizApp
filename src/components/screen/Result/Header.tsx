import * as React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';

interface HeaderProps {
	goBack: () => void;
}

const Header = ({ goBack }: HeaderProps) => (
	// eslint-disable-next-line @typescript-eslint/no-use-before-define
	<View style={styles.container}>
		<TouchableOpacity style={styles.backButton} onPress={goBack}>
			<Text>Back</Text>
		</TouchableOpacity>
		<Text>Result</Text>
	</View>
);

export default Header;

const styles = StyleSheet.create({
	container: {
		width: '100%',
		flexDirection: 'row',
		justifyContent: 'center',

		// backgroundColor: 'red',
		paddingVertical: 10,

		borderBottomColor: 'gray',
		borderBottomWidth: 1,
	},
	backButton: {
		position: 'absolute',
		left: 0,
		paddingTop: 10,
		marginLeft: 16,
	},
});
