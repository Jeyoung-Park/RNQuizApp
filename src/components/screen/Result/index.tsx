import * as React from 'react';
import { Text, View, StyleSheet, Dimensions } from 'react-native';
import { useSelector } from 'react-redux';
import { PieChart } from 'react-native-chart-kit';
import config from '../../../config';

interface ResultProps {
	navigation: any;
}

const Result = ({ navigation }: ResultProps) => {
	const correctNumber = useSelector((state) => state.quiz.correctNumber);
	const quizTime = useSelector((state) => state.quiz.quizTime);
	const quizList = useSelector((state) => state.quiz.quizList.result?.results);

	const data = [
		{
			name: '정답',
			value: correctNumber,
			color: 'rgba(131, 167, 234, 1)',
			legendFontColor: '#7F7F7F',
			legendFontSize: 15,
		},
		{
			name: '오답',
			value: quizList?.length - correctNumber,
			color: '#F00',
			legendFontColor: '#7F7F7F',
			legendFontSize: 15,
		},
	];

	console.log('data in Result index, ', data);

	return (
		// eslint-disable-next-line @typescript-eslint/no-use-before-define
		<View style={styles.container}>
			<Text>Result</Text>
			{data && (
				<PieChart
					data={data}
					width={config.WINDOW_WIDTH}
					height={220}
					// chartConfig={chartConfig}
					accessor="value"
					backgroundColor="transparent"
					paddingLeft="20"
					center={[0, 0]}
					// absolute
					chartConfig={{
						backgroundColor: '#e26a00',
						backgroundGradientFrom: '#fb8c00',
						backgroundGradientTo: '#ffa726',
						decimalPlaces: 2, // optional, defaults to 2dp
						color: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
						labelColor: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
						style: {
							borderRadius: 16,
							// backgroundColor:'pink',
						},
						propsForDots: {
							r: '6',
							strokeWidth: '2',
							stroke: '#ffa726',
						},
					}}
					style={{
						marginVertical: 8,
						borderRadius: 16,
						// backgroundColor: 'purple',
					}}
				/>
			)}
			<Text>전체 문제: {quizList?.length}</Text>
			<Text>정답 개수: {correctNumber}</Text>
			<Text>오답 개수: {quizList?.length - correctNumber}</Text>
			<Text>소요 시간: {quizTime} </Text>
		</View>
	);
};

export default Result;

const styles = StyleSheet.create({
	container: {},
});
