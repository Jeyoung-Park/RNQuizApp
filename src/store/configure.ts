import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../models';

const configure = () => createStore(reducer, applyMiddleware(thunk));

export default configure;
