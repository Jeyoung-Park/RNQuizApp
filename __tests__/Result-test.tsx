import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import store from '../src/store';
import Result from '../src/components/screen/Result';

jest.mock('react-native-simple-toast', () => ({
	SHORT: jest.fn(),
}));

it('snapshot test', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<Result route={{ params: [1, 2, 3] }} />
			</Provider>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
