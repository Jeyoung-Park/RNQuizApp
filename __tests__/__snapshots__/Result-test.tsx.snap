// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Result
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 16,
        "flex": 1,
        "height": 220,
        "justifyContent": "center",
        "marginVertical": 8,
        "padding": 0,
        "width": 750,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={220}
      bbWidth={750}
      focusable={false}
      height={220}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 220,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup>
        <RNSVGGroup>
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  -291840,
                  1,
                  -22746,
                ]
              }
              gradientTransform={null}
              gradientUnits={1}
              name="backgroundGradient"
              x1={0}
              x2={220}
              y1={220}
              y2={0}
            />
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  452984831,
                  1,
                  16777215,
                ]
              }
              gradientTransform={null}
              gradientUnits={1}
              name="fillShadowGradient"
              x1={0}
              x2={0}
              y1={0}
              y2={220}
            />
          </RNSVGDefs>
        </RNSVGGroup>
        <RNSVGRect
          fill={0}
          height={220}
          propList={
            Array [
              "fill",
            ]
          }
          rx={16}
          ry={16}
          width="100%"
          x={0}
          y={0}
        />
        <RNSVGGroup
          matrix={
            Array [
              1,
              0,
              -0,
              1,
              207.5,
              110,
            ]
          }
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M NaN NaN A 88 88 0 0 1 NaN NaN L NaN NaN A 0 0 0 0 0 NaN NaN Z "
              fill={4286818282}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGRect
              fill={4286818282}
              height="16px"
              propList={
                Array [
                  "fill",
                ]
              }
              rx={8}
              ry={8}
              width="16px"
              x={276}
              y={-76}
            />
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={4286545791}
              font={
                Object {
                  "fontSize": 15,
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  300,
                ]
              }
              y={
                Array [
                  -64,
                ]
              }
            >
              <RNSVGTSpan
                content="NaN% 정답"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGPath
              d="M NaN NaN A 88 88 0 0 1 NaN NaN L NaN NaN A 0 0 0 0 0 NaN NaN Z "
              fill={4294901760}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGRect
              fill={4294901760}
              height="16px"
              propList={
                Array [
                  "fill",
                ]
              }
              rx={8}
              ry={8}
              width="16px"
              x={276}
              y={12}
            />
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={4286545791}
              font={
                Object {
                  "fontSize": 15,
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  300,
                ]
              }
              y={
                Array [
                  24,
                ]
              }
            >
              <RNSVGTSpan
                content="NaN% 오답"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
        "paddingBottom": 20,
      }
    }
  >
    <Text>
      전체 문제: 
    </Text>
    <Text>
      정답 개수: 
    </Text>
    <Text>
      오답 개수: 
      NaN
    </Text>
    <Text>
      소요 시간: 
       
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginVertical": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "gray",
          "borderRadius": 20,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 20,
          "width": "80%",
        }
      }
    >
      <Text>
        다시 풀기
      </Text>
    </View>
  </View>
</View>
`;
